specification/maintenance/Maintenance.Management/ConfigurationAssignment.tsp

```diff
@@ -0,0 +1,301 @@
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+
+        /**
+         * Resource provider name
+         */
+        @path @segment("providers") providerName: string,
+
+        /**
+         * Resource parent type
+         */
+        @path resourceParentType: string,
+
+        /**
+         * Resource parent name
+         */
+        @path resourceParentName: string,
+
+        /**
+         * Resource type
+         */
+        @path resourceType: string,
+
+        /**
+         * Resource name
+         */
+        @path resourceName: string,
+
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = ConfigurationAssignment,
+        KeyName = "configurationAssignmentName",
+        SegmentName = "configurationAssignments",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+@armResourceOperations
+interface ConfigurationAssignments {
+  /**
+   * Get configuration assignment for resource..
+   */
+  getParent is ConfigurationAssignmentOps.Read<ConfigurationAssignment>;
+
+  /**
+   * Register configuration for resource.
+   */
+  createOrUpdateParent is ConfigurationAssignmentOps.CreateOrUpdateSync<ConfigurationAssignment>;
+
+  /**
+   * Unregister configuration for resource.
+   */
+  deleteParent is ConfigurationAssignmentOps.DeleteSync<
+    ConfigurationAssignment,
+    Response = ArmResponse<ConfigurationAssignment> | ArmDeletedNoContentResponse
+  >;
+
+  /**
+   * List configurationAssignments for resource.
+   */
+  listParent is ConfigurationAssignmentOps.List<
+    ConfigurationAssignment,
+    Response = ArmResponse<ListConfigurationAssignmentsResult>
+  >;
+}
+@armResourceOperations
+interface ConfigurationAssignmentOperationGroupOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+
+        /**
+         * Resource provider name
+         */
+        @path @segment("providers") providerName: string,
+
+        /**
+         * Resource parent type
+         */
+        @path resourceType: string,
+
+        /**
+         * Resource parent name
+         */
+        @path resourceName: string,
+
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = ConfigurationAssignment,
+        KeyName = "configurationAssignmentName",
+        SegmentName = "configurationAssignments",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+@armResourceOperations
+interface ConfigurationAssignmentOperationGroup {
+  /**
+   * Get configuration assignment for resource..
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("ConfigurationAssignments_Get")
+  get is ConfigurationAssignmentOperationGroupOps.Read<ConfigurationAssignment>;
```

Comment:
Why can't we just put these operations in the previous interface and avoid all of the `@operationId` decorators we will have to replace in the future?